spec/models/search/bookmark_search_form_spec.rb

```diff
@@ -62,13 +62,33 @@
         end
 
         it "changes when the work is updated" do
-          work2.update_attribute(:revised_at, Time.now)
+          work2.update_attribute(:revised_at, Time.current)
           run_all_indexing_jobs
           results = BookmarkSearchForm.new(
             parent: tag, sort_column: "bookmarkable_date"
           ).bookmarkable_search_results
           expect(results.map(&:title)).to eq ["Two", "One", "Three"]
         end
+
+        it "doesn't change tied bookmarkables order on work update" do
+          works = [work1, work2, work3]
+          revised_at = Time.current
+          works.each do |work|
+            work.update_attribute(:revised_at, revised_at)
+          end
+          run_all_indexing_jobs
+          search = BookmarkSearchForm.new(
+            parent: tag, sort_column: "bookmarkable_date"
+          )
+          res = search.bookmarkable_search_results.map(&:title)
+
+          works.each do |work|
+            work.update(summary: "Updated")
+            work.update_attribute(:revised_at, revised_at)
```

Comment:
Why do we set `revised_at` for a second time here? Does it get changed otherwise?

Reply:
It's not really seemed to be needed now, but I added it on purpose.

As I go for update through controller, some functionality regarding updating revised date might be touched. Maybe it won't ever be, maybe it will in some rare cases. If something is changed in the future, I do not want tests to silently become invalid.
As prerequisite for this test is that revised_date is always the same, I want to always assure it explicitly.

Reply:
Gotcha. In that case, instead of setting `revised_at` a second time, let's add an expect that `revised_at` is the same as the initial value.

Reply:
Done, though I think "expects" is about expected outcome of what is tested and "revised_at" is not the point of the test. So I like it a bit less then setting the value.
But as we don't have any other prerequisite assertion, I am fine with it.

Reply:
Yeah, you're right. Plus I forgot how annoying it is to test date time equality instead of freezing time.

Setting the value should be fine, if you add a comment mentioning that `revised_at` should stay the same, and is the sorting option being tested.

Reply:
I think it'd be cleaner to just freeze time. If you place the freeze in the block before the `let!`s, it should mean that you don't have to manually set any dates.

Reply:
I've updated for now with a comment. Not sure how freeze should work.
```
Timecop.freeze do 
 # all the lets and stuff but without any reference to time (no revised_at, no created_at)
```
apparently didn't freeze time without explicitly setting it in lets. So maybe I misunderstood.
Also if it works like that it will clash with bookmarkable_search_result tests and I am already not sure what I would be expected to move where 😅

Reply:
It should be pretty straightforward to freeze time in the `search_results` tests (try calling just `before { freeze_time }`).

As for the tests in this block, I'd be inclined to move them into a separate context where you can freeze time (e.g. `"when there are a variety of times"` vs `"when everything was created at the same time"`). The existing tests -- and all of the objects defined in them -- were designed around the exact opposite of the thing that you're trying to test for, which is contributing to the verboseness of the tests you're writing.

Reply:
Thank you! All good with correct freeze.